### PR TITLE
added option to dump complete raw data to json

### PR DESCRIPTION
**Pull request by Avi-Robusta**
`poetry run python3 holmes.py ask "What is the issue with payment-processing-worker" -raw-json-file="out.json"`

### CODE REVIEW

File: holmes.py [hunk 1]
```diff
@@ -100,6 +100,11 @@ def init_logging(verbose = False):
     "--slack-channel",
     help="Slack channel if --destination=slack (experimental). E.g. #devops",
 )
+opt_dump_raw_output: Optional[str] = typer.Option(
+    None,
+    "-raw-json-file",
```
Comment (aantn) [inline holmes.py]: Can we name it --json-output-file please.
If you name the cli argument `json_output_file` then I don't think you'll have to explicitly specify the cli flag.

File: holmes.py [hunk 2]
```diff
@@ -148,6 +154,9 @@ def ask(
     console.print("[bold yellow]User:[/bold yellow] " + prompt)
     response = ai.call(system_prompt, prompt)
     text_result = Markdown(response.result)
+    if dump_raw_output:
+        with open(dump_raw_output , 'w' , encoding='utf-8') as f:
+            json.dump(json.loads(response.model_dump_json()), f, ensure_ascii=False, indent=4)
```
Comment (aantn) [inline holmes.py]: Do we need the 3-way roundtrip? Is there a reason not to write `response.model_dump_json()` to file?

File: holmes.py [hunk 3]
```diff
@@ -234,11 +244,13 @@ def alertmanager(
         console.print(
             f"[bold yellow]Analyzing all {len(issues)} issues. (Use --alertname to filter.)[/bold yellow] [red]Press Ctrl+C to stop.[/red]"
         )
+    results = []
     for i, issue in enumerate(issues):
         console.print(
             f"[bold yellow]Analyzing issue {i+1}/{len(issues)}: {issue.name}...[/bold yellow]"
         )
         result = ai.investigate(issue, system_prompt, console)
+        results.append({"issue": json.loads(issue.json()), "result": json.loads(result.json())})
```
Comment (aantn) [inline holmes.py]: Here too, I don't think the roundtrip is necessary. What are you trying to achive with it?

File: holmes.py [hunk 4]
```diff
@@ -250,6 +262,9 @@ def alertmanager(
         elif destination == DestinationType.SLACK:
             slack.send_issue(issue, result)
 
+    if dump_raw_output:
```
Comment (aantn) [inline holmes.py]: Can you create a shared function to dump the json that both commands use? (And please add it to the other commands as well for consistency.)

I suspect there is an idiomatic way to dump pydantic models to file that will work and be re-usable in all the calls.

**Review by aantn — APPROVED** [review]
Left some comments.

File: holmes.py (hunk 2 above)
Comment (Avi-Robusta) [reply holmes.py]: model_dump_json makes it a json string, used model_dump

File: holmes.py (hunk 4 above)
Comment (Avi-Robusta) [reply holmes.py]: done

**Review by aantn — APPROVED** [review]

**Pull request merged**